### PR TITLE
drivers: npmx: make interrupt pins optional

### PR DESCRIPTION
**Pull request by jaz1-nordic**
JIRA: NRFX-4670

### CODE REVIEW

File: drivers/npmx/npmx_driver.c [hunk 1]
```diff
@@ -119,14 +119,20 @@ static int int_gpio_interrupt_init(const struct device *dev)
 {
 	struct npmx_data *data = dev->data;
 	const struct npmx_config *config = dev->config;
+
+	if (config->pmic_int_pin != -1) {
```
Comment (wilu-nordic) [inline drivers/npmx/npmx_driver.c:123]: It should me moved to the previous function, the same as here: https://github.com/NordicSemiconductor/npmx-zephyr/pull/77/files#diff-96661c646d9ec2d1a410f047738502496cd688985451a8546a62a3c65fba5919R270

Also clearing interrupts can be skipped.

File: drivers/npmx/npmx_driver.c [hunk 2]
```diff
@@ -246,18 +246,30 @@ static int npmx_driver_init(const struct device *dev)
...
 
-	/* Clear all events before enabling interrupts. */
-	for (uint32_t i = 0; i < NPMX_EVENT_GROUP_COUNT; i++) {
-		if (npmx_core_event_interrupt_disable(npmx_instance, (npmx_event_group_t)i,
-						      NPMX_EVENT_GROUP_ALL_EVENTS_MASK) !=
-		    NPMX_SUCCESS) {
-			LOG_ERR("Failed to disable interrupts");
-			return -EIO;
+	if ((config->host_int_gpio.port != NULL) &&
+		(config->pmic_int_pin != -1))
```
Comment (wilu-nordic) [inline drivers/npmx/npmx_driver.c:250]: Please use clang-format

File: drivers/npmx/npmx_driver.c [hunk 3]
```diff
@@ -246,18 +246,30 @@ static int npmx_driver_init(const struct device *dev)
...
+			LOG_ERR("Failed to initialize interrupt");
+			return -EIO;
+		}
+	} else if ((config->host_int_gpio.port != NULL) &&
+			   (config->pmic_int_pin == -1)){
+		LOG_ERR("PMIC INT pin not configured");
+		return -EIO;
+	} else if ((config->host_int_gpio.port == NULL) &&
+			   (config->pmic_int_pin != -1)){
+		LOG_ERR("HOST INT pin not configured");
```
Comment (wilu-nordic) [inline drivers/npmx/npmx_driver.c:272]: Can be simplified
```suggestion
	} else if (config->pmic_int_pin == -1) {
		LOG_ERR("PMIC INT pin not configured");
		return -EIO;
	} else if (config->host_int_gpio.port == NULL) {
		LOG_ERR("HOST INT pin not configured");
```

**Review by magp-nordic — APPROVED** [review]

**Review by magp-nordic — DISMISSED** [review]

**Review by masz-nordic — APPROVED** [review]

**Comment by nordic-segl** [comment]
Waiting for results from integration CI
https://jenkins-ncs.nordicsemi.no/job/HCS/job/NPMX/job/npmx_zephyr_integration/job/main/436/

**Comment by nordic-segl** [comment]
Waiting for results of integration CI (after PR rebase)
https://jenkins-ncs.nordicsemi.no/job/HCS/job/NPMX/job/npmx_zephyr_integration/job/main/437/

**Review by nordic-segl — APPROVED** [review]

**Review by magp-nordic — APPROVED** [review]

**Pull request merged**